src/parser/select-parser.ts

```diff
@@ -42,11 +42,13 @@ export type SelectArg<
   | ReadonlyArray<SE>
   | ((eb: ExpressionBuilder<DB, TB>) => ReadonlyArray<SE>)
 
-export type Selection<DB, TB extends keyof DB, SE> = {
-  [A in ExtractAliasFromSelectExpression<SE>]: SelectType<
-    ExtractTypeFromSelectExpression<DB, TB, SE, A>
-  >
-}
+export type Selection<DB, TB extends keyof DB, SE> = [SE] extends [unknown]
```

Comment:
is this one of those stack resetting tricks? or just for type errors?

Reply:
Forces the type to be evaluated in-place. You see the resulting type in the IDE instead of `Selection<...>`. Also doesn't increase the stack depth.